dotCMS/src/main/java/com/dotmarketing/servlets/ShortyServlet.java

```diff
@@ -12,177 +12,279 @@
+import com.liferay.util.StringPool;
 
+
+/**
+ * Resolves a shorty or long id, image or assets.
+ * if the path has an jpeg or jpegp would be taken as a image and can resize
+ */
 public class ShortyServlet extends HttpServlet {
 
   private static final long serialVersionUID = 1L;
 
-  final static String FILE_ASSEST_DEFAULT = FileAssetAPI.BINARY_FIELD;
+  private final HostWebAPI     hostWebAPI     = WebAPILocator.getHostWebAPI();
+  private final VersionableAPI versionableAPI = APILocator.getVersionableAPI();
+  private final ShortyIdAPI    shortyIdAPI    = APILocator.getShortyAPI();
+  private final ContentletAPI  contentletAPI  = APILocator.getContentletAPI();
+
+
+  private static final String  JPEG                        = "jpeg";
+  private static final String  JPEGP                       = "jpegp";
+  private static final String  FILE_ASSET_DEFAULT          = FileAssetAPI.BINARY_FIELD;
+  public  static final String  SHORTY_SERVLET_FORWARD_PATH = "shorty.servlet.forward.path";
+  private static final Pattern widthPattern                = Pattern.compile("/\\d+[w]");
+  private static final Pattern heightPattern               = Pattern.compile("/\\d+[h]");
 
-  
-  public final static String SHORTY_SERVLET_FORWARD_PATH = "shorty.servlet.forward.path";
-  
-  
-  protected void service(HttpServletRequest request, HttpServletResponse response)
+  @CloseDBIfOpened
+  protected void service(final HttpServletRequest request, final HttpServletResponse response)
       throws ServletException, IOException {
     try {
-      _serve(request, response);
+
+        serve(request, response);
     } catch (Throwable t) {
       throw new ServletException(t);
-    } finally {
-      DbConnectionFactory.closeSilently();
     }
   }
-  public final String SHORTY_SERVLET_FINAL_PATH = "shorty.servlet.final.path";
 
-  private final static Pattern dimw = Pattern.compile("/\\d+[w]");
-  private final static Pattern dimh = Pattern.compile("/\\d+[h]");
-  private void _serve(final HttpServletRequest request, final HttpServletResponse response) throws Exception {
 
-    Host host = WebAPILocator.getHostWebAPI().getCurrentHost(request);
+  private int getWidth(final String uri, final int defaultWidth) {
 
+    int width = 0;
 
-    PageMode mode = PageMode.get(request);
+    try {
+      final Matcher widthMatcher = widthPattern.matcher(uri);
+      width = widthMatcher.find()?
+              Integer.parseInt(widthMatcher.group().substring(1).replace("w", StringPool.BLANK)):
+              defaultWidth;
+    } catch(Exception e){
+      Logger.debug(this, e.getMessage());
+    }
 
+    return width;
+  }
 
+  private int getHeight (final String uri, final int defaultHeight) {
 
-    // Checking if host is active
-    if (!mode.isAdmin && !APILocator.getVersionableAPI().hasLiveVersion(host)) {
-      response.sendError(HttpServletResponse.SC_SERVICE_UNAVAILABLE,
-          LanguageUtil.get("server-unavailable-error-message"));
```

Comment:
It looks like we lost the host.isLive() check - we should bring this logic back as it is valid

Reply:
It has been moved to the method isValidRequest